File: PythonSandbox/src/leetcode/lc168_excel_column_title.py
'''
https://leetcode.com/problems/excel-sheet-column-title/
'''
import string 

class Solution:
    def convertToTitle(self, n: int) -> str:
        title = ""
        while n > 0:
            #print("n: ", n)
            mod = int(n%26)
            if mod > 0:
                title += string.ascii_uppercase[mod-1]
                n -= mod
            else:
                title += "Z"
                n -= 1
            n = int(n/26)
        
        #print("title: ", title)
        return title[::-1]

File: JavaSandbox/src/leetcode/lc168_ExcelColumnTitle.java
/*
https://leetcode.com/problems/excel-sheet-column-title/
*/
package leetcode;

public class lc168_ExcelColumnTitle {
    public String convertToTitle(int n) {
        
        char[] letters = new char[26];
        for(char c='A'; c <= 'Z'; c++) {
            letters[c-'A'] = c;
        }
        //System.out.println("letters: " + Arrays.toString(letters));
        
        String title = "";
        while(n>0) {
            int mod = n%26;
            if(mod > 0) {
                title += letters[mod-1];
                n -= mod;
            } else {
                title += 'Z';
                n -= 1;
            }
            n = n/26;
        }
        return new StringBuilder(title).reverse().toString();
    }
}
